packages/x-components/src/x-modules/tagging/service/pdp-add-to-cart.service.ts

```diff
@@ -99,29 +103,26 @@ export class DefaultPDPAddToCartService implements PDPAddToCartService {
    *
    * @internal
    */
-  protected getStorageId(id: string): string {
+  protected getStorageId(id?: string): string | null {
     if (this.clickedResultStorageKey === 'url') {
-      const url = id.replace(/\s|\+/g, '%20');
+      const url = id ? id.replace(/\s|\+/g, '%20') : window.location.href.replace(/\s|\+/g, '%20');
```

Comment:
if the `url` is configured to be used, then we use it directly:
`const url = window.location.href.replace(/\s|\+/g, '%20');`